### PR TITLE
Closes #11480: Enable TCP in strict mode and private mode

### LINKED ISSUE #11480: Enable TCP  in strict mode and private mode
Related https://github.com/mozilla-mobile/fenix/pull/22766#pullrequestreview-843649243

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-19045)

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ permalink: /changelog/
 
 * * **browser-engine**, **concept-engine*** **feature-sitepermissions**
   * 🌟️️ **Add support for a new `storage_access` API prompt.
+  * 🌟️ Updated `TrackingProtectionPolicy.cookiePolicyPrivateMode` and `TrackingProtectionPolicy.cookiePolicy` to be `ACCEPT_FIRST_PARTY_AND_ISOLATE_OTHERS` on `TrackingProtectionPolicy.strict()` .
+  * 🌟️ Updated `TrackingProtectionPolicy.cookiePolicyPrivateMode` to be `ACCEPT_FIRST_PARTY_AND_ISOLATE_OTHERS` on `TrackingProtectionPolicy.strict()` .
```
Comment (Mugurell) [inline docs/changelog.md:20]:
```suggestion
  * 🌟️ Updated `TrackingProtectionPolicy.cookiePolicyPrivateMode` to be `ACCEPT_FIRST_PARTY_AND_ISOLATE_OTHERS` on `TrackingProtectionPolicy.recommended()` .
```

**Review by Mugurell — APPROVED** [review]

File: docs/changelog.md (hunk 1 above)
Comment (Amejia481) [reply docs/changelog.md]: Updated, thanks for finding it :)

File: components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt [hunk 2]
```diff
@@ -415,7 +416,8 @@ abstract class EngineSession(
              */
             fun strict() = TrackingProtectionPolicyForSessionTypes(
                 trackingCategory = arrayOf(TrackingCategory.STRICT),
-                cookiePolicy = ACCEPT_NON_TRACKERS,
+                cookiePolicy = ACCEPT_FIRST_PARTY_AND_ISOLATE_OTHERS,
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt:420]: Updated, thanks for finding it :)

**Comment by Amejia481** [comment]
This came from our sprint planning, we want to delay this from landing until we do a further investigation.

**Pull request closed without merging**